Impl [UI] Redesign to the `Navbar` and tabs `1.11.0`

ilan7empest opened the pull request:
- **UI**: Redesign to the `Navbar` and tabs
   Depends On: https://github.com/iguazio/dashboard-react-controls/pull/445
   
   Jira: [ML-11082](https://iguazio.atlassian.net/browse/ML-11082), [ML-11083](https://iguazio.atlassian.net/browse/ML-11083), [ML-11084](https://iguazio.atlassian.net/browse/ML-11084)

   After:
   <img width="291" height="893" alt="Screenshot 2025-10-28 at 11 14 03" src="https://github.com/user-attachments/assets/4304c1fa-01be-4e3c-87bd-0ff9260cf97f" />
   <img width="149" height="311" alt="Screenshot 2025-10-28 at 11 14 56" src="https://github.com/user-attachments/assets/e3b90bf0-34fa-45e4-a0e2-b492d635fc1d" />


[ML-11082]: https://iguazio.atlassian.net/browse/ML-11082?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ML-11083]: https://iguazio.atlassian.net/browse/ML-11083?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ML-11084]: https://iguazio.atlassian.net/browse/ML-11084?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Merge development to branch

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
fix navbar css

ilan7empest committed:
fix navbar expend state

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Impl [Navbar] new look by Figma

ilan7empest committed:
Add alerts item to breadcrumbs

mavdryk commented on src/common/Breadcrumbs/Breadcrumbs.jsx:
```diff
@@ -45,42 +48,43 @@ const Breadcrumbs = ({ onClick = () => {} }) => {
         screen,
-        tab
+        tab: { ...tab, label: startCase(tab?.label?.replace('-', ' ')) || '' },
+        itemName: params.artifactName || params.funcName || params.jobName || params.workflowProjectName || params.appName || params.name || params.alertName || null,
```
These values should be passed via props

mavdryk commented on src/components/FeatureStore/FeatureStorePageTabs/FeatureStorePageTabs.jsx:
```diff
@@ -32,6 +32,7 @@ import {
 } from '../../../constants'
 
 const FeatureStorePageTabs = () => {
+  // To delete after new navbar is merged
```
Not sure why we need this comment. Looks like we still have tabs for Feature Store

mavdryk commented on src/common/Breadcrumbs/Breadcrumbs.jsx:
```diff
@@ -45,42 +48,43 @@ const Breadcrumbs = ({ onClick = () => {} }) => {
+        pathItems: [projects, projectName, screen?.label],
         screen,
-        tab
+        tab: { ...tab, label: startCase(tab?.label?.replace('-', ' ')) || '' },
```
`tab` isn't used, can be deleted

mariana-furyk commented on src/common/Breadcrumbs/BreadcrumbsStep/BreadcrumbsStep.jsx:
```diff
@@ -159,62 +168,84 @@ const BreadcrumbsStep = React.forwardRef(
+    return isLastStep ? (
       <>
-        {isLastStep ? (
+        {urlParts.itemName ? (
```
remove this new part, we can now use pathItem.<id | label | link>

mariana-furyk commented on src/elements/BreadcrumbsDropdown/BreadcrumbsDropdown.jsx:
```diff
@@ -72,8 +75,7 @@ const BreadcrumbsDropdown = forwardRef(
               return (
-                !listItem.hidden &&
-                (listItem.link ? (
+                listItem.link ? (
```
```suggestion
                listItem.externalLink ? (
```

mariana-furyk commented on src/elements/NavbarList/NavbarList.jsx:
```diff
@@ -0,0 +1,173 @@
+    const projectsList = generateProjectsList(projectStore.projectsNames.data)
+    return projectsList.map(project => ({
+      ...project,
+      linkTo: pathname.replace(params.projectName, project.id)
```
```suggestion
      link: pathname.replace(params.projectName, project.id)
```

ilan7empest committed:
Fix by code review

ilan7empest committed:
Fix by code review

ilan7empest committed:
Rename navbarlist.util.jsx to navbarList.util.jsx

ilan7empest committed:
Remove unused package

ilan7empest committed:
Fix missing itemName prop

ilan7empest committed:
Add missing itemName prop

mariana-furyk commented on src/common/Breadcrumbs/Breadcrumbs.jsx:
```diff
@@ -21,66 +21,70 @@ import React, { useMemo, useRef, useState } from 'react'
-        tab = {}
-      }
+    const [projects, projectName, page, innerScreenName] = location.pathname.split('/').slice(1, 5)
+    const screen = mlrunScreens.find(screen => screen.id === (page)) || mlrunScreens.find(screen => screen.id === (innerScreenName))
```
bug with nested screens

mariana-furyk commented on src/common/Breadcrumbs/Breadcrumbs.jsx:
```diff
@@ -21,66 +21,70 @@ import React, { useMemo, useRef, useState } from 'react'
+    const screen = mlrunScreens.find(screen => screen.id === (page)) || mlrunScreens.find(screen => screen.id === (innerScreenName))
+
+    const pathItems = [
+      { id: projects, label: capitalize(projects), link: `/${PROJECTS_PAGE_PATH}` },
```
```suggestion
      { id: projects, label: 'Projects', link: `/${PROJECTS_PAGE_PATH}` },
```

mariana-furyk commented on src/common/Breadcrumbs/Breadcrumbs.jsx:
```diff
@@ -101,7 +106,7 @@ const Breadcrumbs = ({ onClick = () => {} }) => {
               showProjectsList={showProjectsList}
               showScreensList={showScreensList}
-              urlPart={urlPart}
+              pathItem={urlPart}
```
rename urlPart param to "pathItem"

mariana-furyk commented on src/common/Breadcrumbs/BreadcrumbsStep/BreadcrumbsStep.jsx:
```diff
@@ -127,95 +119,94 @@ const BreadcrumbsStep = React.forwardRef(
 
     const handleSeparatorClick = (nextItem, separatorRef) => {
-      const nextItemIsScreen = Boolean(mlrunScreens.find(screen => screen.label === nextItem))
+      if (nextItem.type === 'screen' || nextItem.type === 'project') {
```
use constants instead of strings

mariana-furyk replied:
use constants instead of strings

mariana-furyk commented on src/common/Breadcrumbs/BreadcrumbsStep/BreadcrumbsStep.jsx:
```diff
@@ -127,95 +119,94 @@ const BreadcrumbsStep = React.forwardRef(
+      <li
+        data-testid="breadcrumbs-last-item"
+        className="breadcrumbs__item"
+        key={pathItem.id}
```
if project is equal to screen there will be duplicated key

mariana-furyk commented on src/common/Breadcrumbs/BreadcrumbsStep/BreadcrumbsStep.jsx:
```diff
@@ -127,95 +119,94 @@ const BreadcrumbsStep = React.forwardRef(
-              </RoundedIcon>
-              {showScreensList && urlParts.pathItems[index + 1] === urlParts.screen?.label && (
+          <li key={index} className="breadcrumbs__item">
+            {urlParts.pathItems[index + 1]?.type === BREADCRUMBS_STEP_ITEM_TYPE ? (<div className='breadcrumbs__separator'><ArrowIcon /></div >) : <RoundedIcon
```
move to const urlParts.pathItems[index + 1]

mariana-furyk commented on src/common/Breadcrumbs/BreadcrumbsStep/BreadcrumbsStep.jsx:
```diff
@@ -226,23 +217,21 @@ BreadcrumbsStep.propTypes = {
-  urlPart: PropTypes.string.isRequired,
   urlParts: PropTypes.shape({
-    pathItems: PropTypes.arrayOf(PropTypes.string).isRequired,
+    pathItems: PropTypes.arrayOf(PropTypes.object).isRequired,
```
describe pathItem object type

mariana-furyk commented on src/elements/NavbarList/navbarList.util.jsx:
```diff
@@ -0,0 +1,164 @@
+        {
+          id: MODELS_PAGE,
+          label: 'Model artifacts',
+          link: `${pathname}/${MODELS_PAGE}/models`
```
use const for models and other strings

ilan7empest committed:
Fix by code review

ilan7empest committed:
Fix missing breadcrumbs dropdown margin

ilan7empest committed:
Fix by code review

ilan7empest committed:
Fix lint errors

ilan7empest committed:
Fix lint errors

ilan7empest committed:
align with dev and fix format

illia-prokopchuk committed:
[FIX] LLM prompts missing for non-tagged models (#3527)

    ## 📝 Description
  <!-- A clear, concise summary of what this PR does. -->
  <!-- Include context, motivation, or related issues (e.g., "Replaces Sass with CSS Modules"). -->

---

## 🛠️ Changes Made
  <!-- List the main updates in this PR. -->
<!-- Example:
  - Replaced Sass styles with CSS Modules
  - Updated Button component props for accessibility
  - Adjusted CI workflow to run on push for all branches
  -->

---

## ✅ Checklist

- [x] I have given the PR a well-structured title describing the domain and the specific change that was made
- [x] I tested the changes in the browser (locally or via preview build)
- [ ] I confirmed that existing tests pass
- [ ] I added or updated unit / integration tests (if needed)
- [x] I checked that this change doesn’t introduce new console warnings or lint / formatting errors
- [x] I updated the relevant Jira ticket with the appropriate details and status


---

## 🔗 References
- Related ticket / issue: https://iguazio.atlassian.net/browse/ML-11672
---

## 🚨 Potentially Breaking Changes
- [ ] Yes
- [x] No

<!-- If yes, describe what breaks and how to migrate: -->
<!-- Example: Renamed prop `variant` → `type` in Button component -->

---

Includes DRC change
- [ ] Yes
- [x] No

If yes -> requires bump NPM version

---

## 🔍 Additional Notes
<!-- Optional: other context, performance considerations, or follow-up work -->
<!-- Example:
  - TODO: Migrate remaining components away from Sass
  - Known issue: minor layout flicker on mobile
  -->

---

## 📸 Screenshots / Demos
<!-- Include before/after screenshots, GIFs, or video demos if the change affects UI. -->
<!-- You can drag and drop images here directly in the PR. -->

---

ilan7empest committed:
Update iguazio.dashboard-react-controls dependency `3.2.4-navbar`
